woocommerce/wc-admin
# Fix persisted queries being lost when navigating to excluded page (such as Homescreen)

Fixes woocommerce/woocommerce-admin#6581 

As per the original issue discussion, this PR employs one possible fix for the fact that persisted queries are no longer working as expected in the new navigation feature. 

In short, there was a change made in an [unrelated PR](https://github.com/woocommerce/woocommerce-admin/pull/6247) that resulted in the `NavigationPlugin` component being remounted on every router change, which causes the component to lose state that was being used to implement the persisted queries.

This PR resolves the issue by moving the state around persisted queries in `wp.data`. This allows us to retain the state over router changes, but not over page refreshes (which could get messy). 

### Detailed test instructions:

1. Go to Analytics Report
2. Change Time period
3. Navigate to another report
4. Notice that the time period stays the same
5. Navigate to Homescreen
6. Navigate back to Analytics Report
7. Ensure that the time period is _still_ what you set on step 2
